src/main/java/org/mockito/internal/junit/JUnitTool.java

```diff
@@ -4,15 +4,32 @@
  */
 package org.mockito.internal.junit;
 
+import org.mockito.exceptions.verification.ArgumentsAreDifferent;
+
 public class JUnitTool {
 
-    private static final JUnitDetecter detecter = new JUnitDetecter();
+    private final static boolean hasJUnit = canCreateJUnitClass();
+
+    private JUnitTool() {
+    }
 
-    public static boolean hasJUnit() {
-        return detecter.hasJUnit();
+    /**     
+     * If JUnit is used,  AssertionError is returned that extends from ComparisonFailure      
+     * and hence provide a better IDE support as the comparison result is comparable      
+     */
+    public static AssertionError createArgumentsAreDifferentException(String message, String wanted, String actual) {
+        if (hasJUnit) {
+            return new org.mockito.exceptions.verification.junit.ArgumentsAreDifferent(message, wanted, actual);
```

Comment:
It looks good to me. Yet better have a test with that. @ChristianSchwarz You can look at `ClassLoaders` util class that allows to create custom classloaders.